design-at-cornell/src/explore-courses/title/FilterDropdown.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useState } from "react";
+import { FilterDropdownProps } from "../types/PropertyTypes";
+import {
+    ClearCategoryFilters,
+    FilterDropdownContainer,
+    FilterDropdownScrollContainer,
+    SearchBar
+} from "../ExploreCoursesStyles";
+import FilterCheckbox from "./FilterCheckbox";
+import FilterSelectionBubble from "./FilterSelectionBubble";
+
+const FilterDropdown = ({ checkboxLabels, checkboxData, expand, scroll, onCheck, onClear }: FilterDropdownProps) => {
+    const [query, setQuery] = useState('');
+    const checkboxes = checkboxLabels
+        .filter(label => label.toLowerCase().includes(query))
+        .map(label =>
+            <FilterCheckbox
+                key={label}
+                label={label}
+                checked={checkboxData.has(label)}
+                onClick={() => onCheck(label)}
+            />
+        );
+    console.log(Array.from(checkboxData));
+    const clearFilters = (
+        <ClearCategoryFilters onClick={onClear}>
+            Clear All Filters
+        </ClearCategoryFilters>
+    );
+    const selectionBubbles = checkboxLabels
+        .filter(label => checkboxData.has(label))
+        .map(label =>
+            <FilterSelectionBubble
+                key={label}
+                label={label}
+                color='rgba(175, 174, 230, 0.53)'
+                onClick={() => onCheck(label)}
+            />
+        );
+    if (scroll) return (
+        <FilterDropdownContainer expand={expand}>
+            <SearchBar
+                width='292px'
+                background='#EAEAEA'
+                onChange={e => setQuery(e.currentTarget.value.toLowerCase())}
+            />
+            <FilterDropdownScrollContainer>
+                {checkboxes}
+            </FilterDropdownScrollContainer>
+            {selectionBubbles}
+            {checkboxData.size > 0 && clearFilters}
+        </FilterDropdownContainer>
+    );
+    return (
+        <FilterDropdownContainer expand={expand}>
+            {checkboxes}
+            {selectionBubbles}
```

Comment:
The placement of the selection bubbles is a bit off. Maybe do not include the selection bubbles here where `scroll` is false? When scrolling isn't needed, all labels can be seen at once in the dropdown menu, so I don't think having extra bubbles here is necessary. 

![image](https://user-images.githubusercontent.com/61888193/114293726-66626c00-9a66-11eb-8955-1e61defe8c1a.png)

Reply:
Thanks, I definitely agree! Just made those changes.